include/rocksdb/merge_operator.h

```diff
@@ -27,19 +27,28 @@ class Logger;
 //
 // To use merge, the client needs to provide an object implementing one of
 // the following interfaces:
-//  a) AssociativeMergeOperator - for most simple semantics (always take
-//    two values, and merge them into one value, which is then put back
-//    into rocksdb); numeric addition and string concatenation are examples;
+//  a) IdentityAssocMergeOperator - simplest interface for implementing
+//    merge semantics, always taking two inputs (values or merge operands)
+//    and merging them to one value.
+//    Requires: operator is associative, and no distinction between values
+//    and merge operands.
+//    Examples: numeric addition, and string concatenation.
+//    Non-examples: numeric subtraction (not associative), editable document
+//    (values are documents, merge operands are edits)
 //
 //  b) MergeOperator - the generic class for all the more abstract / complex
```

Comment:
Perhaps consider re-ordering a, b, and c into an order of complexity/powerfulness: e.g.

a) IdentityAssocMergeOperator
b) AssociativeMergeOperator
c) MergeOperator

Reply:
Consider renaming `IdentityAssocMergeOperator`. I think it would be better to have consistent naming, e.g.`IdentityAssociativeMergeOperator` ...or if we really must shorten it, then `IdentAssociativeMergeOperator`